src/utils/randomId.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Creates a random URL-safe string for IDs and unique fallback values.
+ *
+ * @param len Length of the string to generate.
+ */
+export default function randomId(len: number = 21): string {
+	const charset = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_-';
+	let result = '';
+	for (let i = 0; i < len; i += 1) {
+		const randomIndex = Math.floor(Math.random() * charset.length);
```

Comment:
We don't need the security of the Web Crypto API (what nanoid uses), so Math.random seems fine here — it's just for component ids, not cryptographic hashes.